include/libvfio-user.h

```diff
@@ -501,7 +501,13 @@ typedef struct {
     /*
      * Migration state transition callback.
      *
-     * The callback should return -1 on error, setting errno.
+     * The callback should return 0 on success and -errno on error.
+     *
+     * When operating in non-blocking mode (LIBVFIO_USER_FLAG_ATTACH_NB was
+     * passed to vfu_create_ctx) and -EBUSY is returned, transitioning to the
+     * new state becomes asynchronous: libvfio-user does not send a response to
+     * the client and does not process any new messages.  Transitioning to the
```

Comment:
Maybe give a little detail on why this is useful?

Reply:
Done, let me know if you want it rephrased.